substrate/primitives/state-machine/src/basic.rs

```diff
@@ -119,8 +121,9 @@ impl BasicExternalities {
 	}
 }
 
-impl PartialEq for BasicExternalities {
-	fn eq(&self, other: &BasicExternalities) -> bool {
+impl BasicExternalities {
+	/// Same as `Eq` trait but on mutable references.
+	pub fn eq(&mut self, other: &mut BasicExternalities) -> bool {
```

Comment:
```suggestion
	#[cfg(feature = "std")]
	pub fn eq(&mut self, other: &mut BasicExternalities) -> bool {
```
This seems quite expensive.